src/gg20/crypto_tools/k256_serde.rs

```diff
@@ -73,6 +92,9 @@ impl<'de> Visitor<'de> for ScalarVisitor {
                 v.len()
             )));
         }
+
+        // TODO: Check if v is in Z_q before deserializing? k256 doesn't check this (and just does a reduction)
+        // Although it shouldn't be exploitable
```

Comment:
Should we do anything about this? You already have the main check about the length. It seems the k256 code can panic if the length is not appropriate.